### PR TITLE
Set-up test environment

### PR DESCRIPTION
**Pull request by tegsheeb**
In this pull request, the basics for the test environment have setup. 

1. **Webpack and babel** - to check they are working, successfully output bundle.js into client/dist and rendered static index.html with App.jsx.
2. **Jest** - installed with babel, webpack, jest-test-render, and their necessary config files. Tested with sample test from the Doc. There are 2 sample tests, one for jest and one for jest-test-render. For jest-test-render, I am not sure we are using it. But according to the Doc, it might be useful for simple tests. 
3. **Enzyme** -installed together with the jest-enzyme environment as the Doc suggested.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        // options: {
+        //   presets: ['react', 'env']
+        // }
+      }
+    ]
+  }
+};
+
+
+// module.exports = {
```
Comment (anaveragejosef) [inline webpack.config.js]: @tegsheeb Do you still need this portion? (Line 27 Onwards)

File: test/react_test/__snapshots__/Link.react.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,34 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (anaveragejosef) [inline test/react_test/__snapshots__/Link.react.test.js.snap:1]: @tegsheeb Is this example data? Or are these results from your tests?

Comment (tegsheeb) [reply test/react_test/__snapshots__/Link.react.test.js.snap:1]: This is the test result from sample test Link.react.test.js. In jest, react-test-renderer takes the snapshot (like in this example) in the end-to-end test and output as a result. When any changes are made to the code being test, jest compares it with the previous snapshot.

File: webpack.config.js (hunk 1 above)
Comment (tegsheeb) [reply webpack.config.js]: Thanks for pointing it out. No, we don't need it. Just cleaned up now.

**Review by anaveragejosef — APPROVED** [review]
Everything looks good!

**Pull request merged**